step-plans/step-plans-core/src/main/java/step/core/execution/ExecutionContext.java

```diff
@@ -225,4 +230,20 @@ public Set<ExecutionVetoer> getExecutionVetoers() {
 	public void addExecutionVetoer(ExecutionVetoer executionVetoer) {
 		this.executionVetoers.add(executionVetoer);
 	}
+
+	public void addAgentUrl(String agentUrl) {
+		if (agentUrl != null) {

```

Comment:
AgentUrls is declared as final and should never be null

Reply:
Nope, that's the entry which is to be added to the set :) . In principle, it *should* never be null if only non-local agents are added, but better safe than sorry, because otherwise it might throw an NPE.

Reply:
My bad, you're right. Was not completely awake :)